orangecanvas/scheme/readwrite.py

```diff
@@ -818,6 +821,9 @@ def check(obj):
                             "literal".format(type(obj)))
 
     def check_relaxed(obj):
+        if type(obj) == float and obj != obj:
+            raise TypeError("NaN can not be serialized as a python literal")
```

Comment:
```suggestion
        if isinstance(obj, numbers.Real) and not math.isfinite(obj):
            raise TypeError("Non-finite values can not be serialized as a python literal")
```
... to allow `numpy.float32`, ...